src/device/asgard_kronmult_cycle1.hpp

```diff
@@ -0,0 +1,324 @@
+#pragma once
+
+#include "asgard_kronmult_common.hpp"
+
+namespace asgard::kronmult::kernel
+{
+/*!
+ * \brief Recursive template that computes n to power, e.g., ipow<2, 3>() returns constexpr 8.
+ *
+ * This is on-device implementation, not really a device function
+ * since it will be evaluated at compile time, but it still
+ * had to be marked as __device__
+ */
+template<int n, int power>
+__device__ constexpr int int_pow()
+{
+  if constexpr (power == 1)
+  {
+    return n;
+  }
+  else if constexpr (power == 2)
+  {
+    return n * n;
+  }
+  else if constexpr (power == 3)
+  {
+    return n * n * n;
+  }
+  else if constexpr (power == 4)
+  {
+    return n * n * n * n;
+  }
+  else if constexpr (power == 5)
+  {
+    return n * n * n * n * n;
+  }
+  else if constexpr (power == 6)
+  {
+    return n * n * n * n * n * n;
+  }
+  else
+  {
+    static_assert(power >= 1 and power <= 6,
+                  "int_pow() does not works with specified power");
+    return 0;
+  }
```

Comment:
I _think_ this will silence some of the warnings from nvcc :crossed_fingers: 
```suggestion
  static_assert(power >= 1 and power <= 6,
                "int_pow() does not works with specified power");
  return 0;
```

Reply:
I'll give it a try, but only enough CUDA 11.8 and newer does not generate the warnings.

Reply:
Fixed that and another one.